Added 4 additional modules to the Cellprofiler tool

sunyi000 opened the pull request:
- added Tile, ColorToGray, TrackObjects and OverlayOutlines modules 
- fixed flake8 linting issue for all python scripts

sunyi000 commented:
@beatrizserrano just open the pr first and I'm still planning to add more tests to the new modules next week..

Cheers

sunyi000 commented:
@beatrizserrano @bgruening this is ready for review. 
thanks

beatrizserrano reviewed (commented):
Thanks a lot, @sunyi000! It has been hard for me to find things to improve here :) Most of my comments are due to the fact that we need users to input the name of images/objects rather than selecting from a drop-down menu. I might have missed some... so bitte double-check :)

beatrizserrano commented on tools/cellprofiler/tile.xml:
```diff
@@ -53,7 +53,7 @@
     <inputs>
         <expand macro="input_pipeline_param"/>
-        <param name="input_image" type="text" value="DNA" label="Select an input image" help="Image to be tiled, specified in NameAndTypes" />
+        <param name="input_image" type="text" value="DNA" label="Input an input image" help="Image to be tiled, specified in NameAndTypes" />
```
```suggestion
        <param name="input_image" type="text" value="DNA" label="Input the name of an input image" help="Image to be tiled, specified in NameAndTypes" />

```

sunyi000 commented:
@beatrizserrano thanks for the review. I believe i have addressed all your comments, except the cdata

the help text is in xml attribute. CDATA can be used on element but not attribute value..

thanks

beatrizserrano commented on tools/cellprofiler/color_to_gray.xml:
```diff
@@ -0,0 +1,282 @@
+            <expand macro="text_validator" />
+        </param>
+        <conditional name="con_conversion_method">
+            <param name="conversion_method" type="select" label="Conversion method" help="How do you want to convert the color image? &lt;br/&gt;
```
https://github.com/bgruening/galaxytools/blob/master/tools/cellprofiler/identify_primary_objects.xml#L15

sunyi000 replied:
ah.. ok I will change them

beatrizserrano commented on tools/cellprofiler/overlay_outlines.xml:
```diff
@@ -0,0 +1,179 @@
+                </param>
+                <when value="Color">
+                    <repeat name="rpt_obj_to_display" title="Outline" min="1">
+                        <param name="obj_to_display" type="text" label="Input objects to display" help="Choose the objects whose outlines you would like to display. "/>
```
```suggestion
                        <param name="obj_to_display" type="text" label="Enter the name of the objects to display" help="Choose the objects whose outlines you would like to display. "/>

there are more like this @sunyi000 
```

beatrizserrano commented on tools/cellprofiler/tile.xml:
```diff
@@ -0,0 +1,139 @@
+    </configfiles>
+    <inputs>
+        <expand macro="input_pipeline_param"/>
+        <param name="input_image" type="text" value="DNA" label="Input the name of an input image" help="Image to be tiled, specified in NameAndTypes" />
```
actually we used sth different before:
```suggestion
        <param name="input_image" type="text" value="DNA" label="Enter the name of an input image" help="Image to be tiled, specified in NameAndTypes" />

```

beatrizserrano reviewed (approved):
This looks great @sunyi000, thanks for the super quick process and adopting all the picky suggestions 'gerne' ;) We are ready to create a super cool workflow now!
